### PR TITLE
[Customer Portal][FE][Web] Integrate Project, Support, and Case API Endpoints with Enhanced Error Feedback

### PR DESCRIPTION
**Pull request by dileepapeiris**
## Purpose
This PR represents a major integration milestone, bridging the gap between static UI components and live backend data. It implements real-time data fetching across the **Project Hub**, **Project Details**, and **Support/Case Management** modules. By integrating six key API endpoints, the application now provides accurate, live metrics while maintaining high visual stability through advanced error handling and data sanitization.

## Goals
- **Full API Integration:** Connect the frontend to live endpoints for Project Stats, Project Details, Case Details, Support Stats, Case Stats, and Case Filters.
- **Data Accuracy & Cleanliness:** Display real-time SLA status, open cases, and active chats, while ensuring backend-supplied descriptions are cleansed of HTML artifacts.
- **Resilient UX:** Implement a standardized "Fail-safe" UI pattern using skeleton loaders for transitions and `ErrorIndicator` components for failed API calls.
- **Build Stability:** Resolve critical TypeScript build blockers (TS6133, TS6196) to ensure a clean production deployment pipeline.

## Approach
- **Standardized Data Fetching:** Utilized specialized hooks (e.g., `useGetProjectStat`, `useGetCaseFilters`, `useGetSupportStats`) across sub-components to allow for decentralized loading and error states.
- **Error Feedback System:** 
    - Integrated `ErrorIndicator` components (with informative tooltips) into dense UI areas like `ProjectCard` stats and `CasesTable` rows.
    - Updated the `ProjectSwitcher` in the header to gracefully handle fetch failures.
- **Content Sanitization:** Developed a `stripHtmlTags` utility in `src/utils/projectCard.ts` to ensure descriptions from the CMS/Backend are rendered as clean, plain text.
- **Visual Stability:**
    - Orchestrated `Skeleton` loaders to match the exact geometry of charts, stats, and badges.
    - Enhanced the `SubscriptionDetails` progress bar to reflect error states when critical date data is missing.
- **Code Health:** Systematic removal of unused variables and imports in `Header.test.tsx` and `mockData.ts`.

## User stories
- **Real-time Monitoring:** As a user, I want to see the exact number of open cases and my current SLA status on the Project Hub so I can prioritize my work immediately.
- **Granular Filtering:** As a user, I want to see live case filters so I can accurately sort through my support history based on real backend categories.
- **Graceful Failure:** As a user, if a specific statistic fails to load, I want to see a clear "Failed to load" indicator rather than a broken or empty UI.
- **Clean Content:** As a user, I want to read project descriptions without seeing technical HTML tags like `<p>` or `<div>`.

## Walkthrough
This PR integrates six core API endpoints across the Project and Support modules. It introduces a description-cleaning utility, implements standardized skeleton/error UI patterns for all new data points, and cleans up the codebase to resolve TypeScript build errors.

## Changes

| Cohort | File(s) | Summary |
| :--- | :--- | :--- |
| **Project Integration** | `ProjectCard.tsx`, `ProjectCardStats.tsx`, `ProjectDescription.tsx` | Integrated `getProjectStat` and `getProjectDetails`; implemented HTML stripping and live metric rendering. |
| **Support Integration** | `CasesTable.tsx`, `useGetSupportStats.ts`, `useGetCaseFilters.ts` | Integrated case stats, filters, and support overview endpoints with error feedback. |
| **Utilities** | `projectCard.ts` | Added `stripHtmlTags` to cleanse backend string data. |
| **UI Components** | `SubscriptionDetails.tsx`, `ProjectCardBadges.tsx` | Added error-state styling and skeleton loaders for real-time status updates. |
| **Build & Maintenance** | `mockData.ts`, `Header.test.tsx` | Resolved TS6133 and TS6196 errors by removing unused code. |
| **Header Suite** | `ProjectSwitcher.tsx`, `Header.tsx` | Synchronized project fetching with auth states and added error icons for fetch failures. |

## Automation tests
 - **Component Testing:** 
   - **ProjectCard/SupportStats:** Verified that skeletons show during fetch and `ErrorIndicator` triggers on 4xx/5xx responses.
   - **ProjectDescription:** Confirmed that descriptions containing HTML tags are rendered as plain text.
 - **Unit Testing:** 
   - Verified `stripHtmlTags` utility against various edge-case strings.
   - Tested Case Filter mapping to ensure backend values correctly populate the UI.
 - **Verification:** 
   - `pnpm build` completed successfully.
   - All 15+ updated test suites passed.

## Summary by CodeRabbit

* **New Features**
  * Dynamic case filter metadata now drives filter options.
  * Dashboard and project UIs surface error states with visual indicators.

* **Improvements**
  * More granular case status and priority labels.
  * Skeleton loaders for clearer loading feedback.
  * Charts and stats show distinct error visualizations; trend/legend reflect error state.
  * Safer null handling with consistent "--" fallbacks across lists and cards.

* **Tests / Mocks**
  * Expanded test coverage and richer UI mocks to validate loading/error flows.

### CODE REVIEW

File: apps/customer-portal/webapp/src/components/dashboard/casesTable/CasesTable.tsx [hunk 1]
```diff
@@ -30,10 +34,58 @@ interface CasesTableProps {
 
 const CasesTable = ({ projectId }: CasesTableProps): JSX.Element => {
   const navigate = useNavigate();
+  const { isLoading: isAuthLoading } = useAsgardeo();
+  const { showLoader, hideLoader } = useLoader();
   const [filters, setFilters] = useState<Record<string, any>>({});
   const [isFilterOpen, setIsFilterOpen] = useState(false);
   const [page, setPage] = useState(0);
-  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const [rowsPerPage, setRowsPerPage] = useState(5);
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/components/dashboard/casesTable/CasesTable.tsx]: -1

File: apps/customer-portal/webapp/src/components/projectDetails/projectOverview/projectInformation/SubscriptionDetails.tsx [hunk 2]
```diff
@@ -77,7 +82,11 @@ const SubscriptionDetails = ({
             variant="determinate"
             value={progress}
             color={
-              subscriptionColor === "default" ? "primary" : subscriptionColor
+              isError || startDate === "--" || endDate === "--"
+                ? "error"
+                : subscriptionColor === "default"
+                  ? "primary"
+                  : subscriptionColor
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/components/projectDetails/projectOverview/projectInformation/SubscriptionDetails.tsx]: Updated

File: apps/customer-portal/webapp/src/components/dashboard/casesTable/CasesTable.tsx [hunk 3]
```diff
@@ -60,7 +112,20 @@ const CasesTable = ({ projectId }: CasesTableProps): JSX.Element => {
...
+    isError,
+  } = useGetProjectCases(projectId, requestBody);
+
+  const tableLoading = isAuthLoading || isFetchingCases || isFetchingFilters;
+
+  useEffect(() => {
+    if (tableLoading) {
+      showLoader();
+      return () => hideLoader();
+    }
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/components/dashboard/casesTable/CasesTable.tsx]: Updated

File: apps/customer-portal/webapp/src/components/projectDetails/projectOverview/projectInformation/SubscriptionDetails.tsx [hunk 4]
```diff
@@ -28,24 +28,27 @@ import {
...
+  isError,
 }: SubscriptionDetailsProps): JSX.Element => {
-  const subscriptionStatus = getSubscriptionStatus(endDate);
+  const subscriptionStatus = getSubscriptionStatus(endDate || "");
   const subscriptionColor = getSubscriptionColor(subscriptionStatus);
   const progress =
     subscriptionStatus === SUBSCRIPTION_STATUS.EXPIRED
       ? 100
-      : calculateProgress(startDate, endDate);
+      : calculateProgress(startDate || "", endDate || "");
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/components/projectDetails/projectOverview/projectInformation/SubscriptionDetails.tsx]: Updated

File: apps/customer-portal/webapp/src/api/__tests__/useGetProjectCases.test.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
+import { renderHook } from "@testing-library/react";
```
Comment (Rashmika998) [inline apps/customer-portal/webapp/src/api/__tests__/useGetProjectCases.test.tsx]: Missing license header

File: apps/customer-portal/webapp/src/components/common/header/__tests__/ProjectSwitcher.test.tsx [hunk 6]
```diff
@@ -56,11 +56,22 @@ vi.mock("@wso2/oxygen-ui", () => ({
   Header: {
     Switchers: ({ children }: any) => <div>{children}</div>,
   },
+  colors: {
+    blue: { 700: "#1d4ed8" },
```
Comment (Rashmika998) [inline apps/customer-portal/webapp/src/components/common/header/__tests__/ProjectSwitcher.test.tsx]:
```suggestion
    blue: { 700: "#1D4ED8" },
```
Let's keep these consistent. You can address this in a later PR

**Review by Rashmika998 — APPROVED** [review]
LGTM
Let's fix the minor changes in a later PR. Please add the issue to track them.

File: apps/customer-portal/webapp/src/utils/projectStats.ts [hunk 7]
```diff
@@ -53,7 +53,10 @@ export const getSLAStatusColor = (
 ): "success" | "error" | "default" | "warning" => {
   const normalizedStatus = status?.toLowerCase();
 
-  if (normalizedStatus === SLA_STATUS.GOOD.toLowerCase()) {
+  if (
+    normalizedStatus === SLA_STATUS.GOOD.toLowerCase() ||
+    normalizedStatus === "good"
```
[the remark this answers is not in the document]
Comment (dileepapeiris) [reply apps/customer-portal/webapp/src/utils/projectStats.ts]: Addressed in https://github.com/wso2-open-operations/cs-tools/pull/95

**Comment by dileepapeiris** [comment]
> LGTM Let's fix the minor changes in a later PR. Please add the issue to track them.

Those changes were addressed via https://github.com/wso2-open-operations/cs-tools/pull/94 and https://github.com/wso2-open-operations/cs-tools/pull/95
